File: fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int) : float =
    match speed with
    | 0 -> 1
    | s when s < 5 -> 1
    | s when s < 9 -> 0.90
    | 9 -> 0.80
    | 10 -> 0.77
    | _ -> failwith "The dial only goes to 10"

let productionRatePerHour (speed: int) : float = 221.0 * float speed * successRate speed

let workingItemsPerMinute (speed: int) : int = int (productionRatePerHour speed) / 60

File: csharp/cars-assemble/CarsAssemble.cs
static class AssemblyLine
{
    static readonly int CARS_PER_HOUR = 221;

    public static double SuccessRate(int speed)
    {
        if (speed == 0)
        {
            return 0;
        }
        else if (speed < 5)
        {
            return 1;
        }
        else if (speed < 9)
        {
            return 0.9;
        }
        else if (speed < 10)
        {
            return 0.8;
        }
        return 0.77;
    }

    public static double ProductionRatePerHour(int speed)
    {
        return speed * CARS_PER_HOUR * SuccessRate(speed);
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        return (int)ProductionRatePerHour(speed) / 60;
    }
}
